tournaments/index.js

```diff
@@ -275,7 +275,7 @@ class Tournament {
 			return;
 		}
 
-		if (this.checkBanned(user)) {
+		if (this.checkBanned(user) || Punishments.isBattleBanned(user)) {
```

Comment:
What about if someone is already in a tournament and they get battle banned while in the middle of it?

Reply:
“Prevents the user from starting new battles”  they would be able to finish the match just like how they would be able to if they got tournament banned while in a tournament match

Reply:
Oh it appears banned users would get kicked, whoops I missed that. Anyways we could just remove the removebanneduser check in checkbanned and move it to this if statement

Reply:
(Feel free to snipe, I’m just about to start my 10 hour shift)

Reply:
It's a lot more complicated than adding it to this if statement.

You'd have to loop thru all games a user is in (``room.games``), and then check which are tournaments (which I don't actually see a way to do.) Then, if its a tournament, you'd have to create a new function to disqualify the user from it, or I guess you could have them parse ``/tour leave``, but that'd be unreliable if they go offline. 

So yeah, this will be more complicated than a one-liner in an if statement. It should probably be done before the server restarts when this code gets live, though.

Reply:
Yeah I already figured that out but ty

Reply:
https://github.com/Zarel/Pokemon-Showdown/pull/4468

Reply:
Completely forgot this still needed fixing on main, ty